skia/RSkComponentBVLinearGradient.cpp

```diff
@@ -0,0 +1,131 @@
+  SharedColor bgColour = linearGradientProps.backgroundColor;
+  SkScalar frameX = frame.origin.x;
+  SkScalar frameY = frame.origin.y;
+  SkScalar frameWidth = frame.size.width;
+  SkScalar frameHeight = frame.size.height;
+
+  // If background color is transparent then need to find atleast one non-transparent colour in gradient colours so that we can draw shadow properly.
+  for(int i = 0; (!(bgColour && (colorComponentsFromColor(bgColour).alpha != 0.0)) && i < numColors); i++) {
+    bgColour = linearGradientProps.colors.at(i);
+  }
+
+  /*Draw Order : 1. Shadow 2. Gradient 3 Border*/
+  auto layerRef=layer();
+  if(layerRef->isShadowVisible) {
+    drawShadow(canvas,frame,borderMetrics,
+                bgColour, // Used only to decide on what type of shadow to draw.
+                layerRef->shadowColor,layerRef->shadowOffset,layerRef->shadowOpacity,
+                layerRef->opacity,
+                layerRef->shadowImageFilter,layerRef->shadowMaskFilter);
+  }
+
+  RNS_LOG_DEBUG(std::this_thread::get_id() << "] Layer " << layer().get() <<" Draw Picture Here with Frame  [" << frameX << "," << frameY << "," << frameWidth << "," << frameHeight << "]");
+
+  SkColor colors[numColors];
+  SkScalar pos[linearGradientProps.locations.size()];
+
+  bool locationsMapAvailable = (linearGradientProps.locations.size() == numColors);
+
+  for (std::size_t i = 0; i != numColors; ++i) {
+    colors[i] = RSkColorFromSharedColor(linearGradientProps.colors.at(i), SK_ColorTRANSPARENT);
+    if(locationsMapAvailable) {
+      pos[i] = linearGradientProps.locations.at(i);
+    }
+  }
+
+  SkPoint startEndPoints[] = { //Initialize with default value
+    {(0.5 * frameWidth) + frameX, (0 * frameHeight) + frameY},
+    {(0.5 * frameWidth)+ frameX, (1 * frameHeight) + frameY}};
+
+  SkMatrix matrx = SkMatrix::I();
+
+  if(linearGradientProps.useAngle) {
+    // Apply angle and center to matrix
+    matrx.setRotate(linearGradientProps.angle + RNS_BVLG_CARTESIAN_ANGLE_CORRECTION,
+                    (linearGradientProps.angleCenter.x * frameWidth)  + frameX,
+                    (linearGradientProps.angleCenter.y * frameHeight) + frameY);
+    // Use default startEnd points
+  } else {
+    startEndPoints[0].set(((linearGradientProps.start.x * frameWidth) + frameX), ((linearGradientProps.start.y * frameHeight) + frameY));
+    startEndPoints[1].set(((linearGradientProps.end.x * frameWidth) + frameX), ((linearGradientProps.end.y * frameHeight) + frameY));
+  }
+
+  auto lgs = SkGradientShader::MakeLinear(startEndPoints,
+                                          colors,
+                                          (locationsMapAvailable ? pos : nullptr),
+                                          numColors,
+                                          SkTileMode::kClamp,
+                                          0/*flags*/, &matrx);
+
+  SkPaint paint;
+  paint.setShader(lgs);
+  canvas->drawRoundRect({frameX, frameY, frameX+frameWidth, frameY+frameHeight},
```

Comment:
If View can take radii on all four Sides, Then drawRRect would be the Right Choice.

Reply:
Done